packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -220,6 +220,7 @@ export async function handleRoute({
 	let response = await pipeline.renderRoute(renderContext, mod);
 	if (response.status === 404 && has404Route(manifestData)) {
 		const fourOhFourRoute = await matchRoute('/404', manifestData, pipeline);
+		// if (fourOhFourRoute?.route !== options.route)
```

Comment:
Commenting out the fix to see what happens to CI.

Reply:
CI times out. Logs for `astro:test` [never appear](https://github.com/withastro/astro/actions/runs/6292593299/job/17092598249?pr=8648#step:9:747). It wouldn't be possible to tell if the cases failed.